README.md

```diff
@@ -1,2 +1,266 @@
+
+To use the extension, the user would load the component to get access to a `LogFn` that can be used in a sequence to log information. A Mixin allows the user to set the application wide logLevel. Only Controller methods configured at or above the logLevel will be logged. 
+
+Possible levels are: DEBUG < INFO < WARN < ERROR < OFF
+
+*Possible levels are represented as numbers but users can use `LOG_LEVEL.${level}` to specify the value instead of using numbers.*
+
+A decorator is provided to allow a user to provide metadata for Controller methods by using it to set the minimum logLevel the method should be logged at.
+
+### Example Usage
+```ts
+import {LogLevelMixin, LogComponent, LOG_LEVEL, log} from 'loopback4-example-log-extension';
+// Other imports ... 
+
+class LogApp extends LogLevelMixin(Application) {
+  constructor() {
+    super({
+      components: [RestComponent, LogComponent],
+      logLevel: LOG_LEVEL.ERROR,
+      controllers: [MyController]
+    });
+  };
+}
+
+class MyController {
+  @log(LOG_LEVEL.WARN)
+  @get('/')
+  hello() {
+    return 'Hello LoopBack';
+  }
+
+  @log(LOG_LEVEL.ERROR)
+  @get('/name')
+  helloName() {
+    return 'Hello Name'
+  }
+}
+```
+
+## Tutorial
+
+We'll start by cloning [loopback4-extension-starter]() and updating files headers to be `loopback4-example-log-extension` and `package.json`.
+
+We'll be keeping all the top level files but modify the `src` directory and `test` directory as follows.
+
+### `/src/index.ts`
+
+We'll update the exports accordingly here.
+
+### `/src/keys.ts`
+
+We'll start by defining new `Binding` values as well as exporting an `enum` for user's to use for setting logLevel. The file will look as follows: 
+
+```ts
+export namespace EXAMPLE_LOG_BINDINGS {
+  export const METADATA = 'example.log.metadata';
+  export const APP_LEVEL = 'example.log.level';
+  export const TIMER = 'example.log.timer';
+  export const LOG_ACTION = 'example.log.action';
+}
+
+export enum LOG_LEVEL {
+  DEBUG,
+  INFO,
+  WARN,
+  ERROR,
+  OFF,
+}
+```
+
+### `/src/log.component.ts`
+
+We'll be keeping this file the same but updating the `Bindings` according to the values we declared in `/src/keys.ts`. This is where we export our providers so they can be automatically be bound to the appropriate keys. The final version of this file will look as follows:
+
+```ts
+import {EXAMPLE_LOG_BINDINGS} from './keys';
+import {Component, ProviderMap} from '@loopback/core';
+import {TimerProvider, LogActionProvider, LogLevelProvider} from './';
+
+export class LogComponent implements Component {
+  providers?: ProviderMap = {
+    [EXAMPLE_LOG_BINDINGS.TIMER]: TimerProvider,
+    [EXAMPLE_LOG_BINDINGS.LOG_ACTION]: LogActionProvider,
+    [EXAMPLE_LOG_BINDINGS.APP_LEVEL]: LogLevelProvider,
+  };
+}
+```
+
+### `/src/types.ts`
+
+We'll be updating 
```

Comment:
A bit awkward to have incomplete sentence. 
Suggestions: 
- We'll be updating the following: ; or
- We'll be updating decorator, mixin and providers